lib/redcord/relation.rb

```diff
@@ -68,15 +80,26 @@ def select(*args, &blk)
 
   sig { returns(Integer) }
   def count
+    model.validate_index_attributes(query_conditions.keys, index_name: index_name)
     redis.find_by_attr_count(
       model.model_key,
       query_conditions,
       index_attrs: model._script_arg_index_attrs,
       range_index_attrs: model._script_arg_range_index_attrs,
+      custom_index_attrs: model._script_arg_custom_index_attrs[index_name],
       hash_tag: extract_hash_tag!,
+      index_name: index_name
     )
   end
 
+  sig { params(custom_index_name: T.nilable(Symbol)).returns(Redcord::Relation) }
+  def with_index(custom_index_name)
```

Comment:
Is with_index always expected to come last? I wonder if we should just implicitly call execute_query here then, or is a query like `TestModel.where(a: val).with_index(:first).where(t: interval)` supposed to work?

Reply:
It is not supposed to always come last. I made sure the case you mentioned is supported. Also you can do:
```
      rel = klass.where(indexed_value: 5, other_value: nil)
      rel2 = rel.with_index(:first)
      rel3 = rel2.with_index(:second)
```